### PR TITLE
INET data type

### LINKED ISSUE #2016: inet data types
Background: https://airtable.com/tblD3oZPLJgGhCmch/viwINAgXA5on9pr2T/rec591c0mW53jVnYZ

PRs: 
- [sql: Add INET column type and IPAddr datum](https://github.com/cockroachdb/cockroach/pull/18171)
- [sql: Add INET type built-in functions](https://github.com/cockroachdb/cockroach/pull/18585)

### PR DESCRIPTION
**Pull request by lhirata**
Closes #2016.

### CODE REVIEW

File: v2.0/inet.md [hunk 1]
```diff
@@ -1,6 +1,86 @@
...
+summary: The INET data type stores an IPv4 or IPv6 address.
 toc: false
 ---
-<span class="version-tag">New in v2.0:</span> Documentation for the `INET` datatype coming soon!
+<span class="version-tag">New in v2.0:</span> The `INET` [data type](data-types.html) stores an IPv4 or IPv6 address.
+
+<div id="toc"></div>
+
+## Syntax
+
```
Comment (lhirata) [inline v2.0/inet.md]: @nvanbenschoten - how do you determine if a data type can be an interpreted literal, an annotated string literal, or coerced to type INET? Do any of these apply to INET?

File: v2.0/inet.md [hunk 2]
```diff
@@ -1,6 +1,86 @@
...
+An `INET` value can be expressed using the following formats:
+
+Format | Description
+-------|-------------
+IPv4 | Standard [RFC791](https://tools.ietf.org/html/rfc791)-specified format of 4 octets expressed individually in decimal numbers and separated by periods. Optionally, the address can be followed by a subnet mask.<br><br> Examples: `'190.0.0.0'`, `'190.0.0.0/24'`
+IPv4-mapped IPv6 | IPv6 address mapped to a IPv4 address. <br><br>Example: `'::ffff:192.168.0.1/24'`
```
Comment (lhirata) [inline v2.0/inet.md]: @nvanbenschoten - do you think this should this be called out as its own format? Or should it be included as an option in the IPv6 description?

File: v2.0/inet.md [hunk 3]
```diff
@@ -1,6 +1,86 @@
...
++--------------------------------------+--------------------------+---------------------------+
+|                  ip                  |        user_email        |     registration_date     |
++--------------------------------------+--------------------------+---------------------------+
+| 192.168.0.1                          | info@cockroachlabs.com   | 2018-01-31 00:00:00+00:00 |
+| 192.168.0.2/10                       | lauren@cockroachlabs.com | 2018-01-31 00:00:00+00:00 |
+| 2001:4f8:3:ba:2e0:81ff:fe22:d1f1/120 | test@cockroachlabs.com   | 2018-01-31 00:00:00+00:00 |
++--------------------------------------+--------------------------+---------------------------+
+~~~
+
+## Supported Casting & Conversion
```
Comment (lhirata) [inline v2.0/inet.md]: @nvanbenschoten - similar to above, how do you determine what INET can be cast to?

File: v2.0/inet.md (hunk 1 above)
Comment (nvanbenschoten) [reply v2.0/inet.md]: EDIT: the first one doesn't actually look like it works.. I'll have to fix that during my QA process (https://github.com/cockroachdb/cockroach/issues/22390). I think it's safe to assume that all three will work by the time this is published.

File: v2.0/inet.md (hunk 3 above)
Comment (nvanbenschoten) [reply v2.0/inet.md]: I think the easiest way is to look at the code: https://github.com/cockroachdb/cockroach/blob/94a2182a5e00fc0963c0f20c1dc278b12a9df7d2/pkg/sql/sem/tree/expr.go#L1254.

`inetCastTypes` is a list of all the types that INET can be cast to. It looks like only STRINGs can be cast to INET. I'll be auditing all of this as part of my QA process.

File: v2.0/data-types.md [hunk 4]
```diff
@@ -10,6 +10,7 @@ CockroachDB supports the following data types. Click a type for more details.
 
 Type | Description | Example
 -----|-------------|--------
+[`INET`](inet.html) | A IPv4 or IPv6 address. | `192.168.0.1`
```
[the remark this answers is not in the document]
Comment (lhirata) [reply v2.0/data-types.md]: Added.

File: v2.0/inet.md (hunk 1 above)
Comment (lhirata) [reply v2.0/inet.md]: Added.

File: v2.0/inet.md (hunk 2 above)
Comment (lhirata) [reply v2.0/inet.md]: Done

File: v2.0/inet.md (hunk 3 above)
Comment (lhirata) [reply v2.0/inet.md]: Thanks!

File: v2.0/sql-constants.md [hunk 5]
```diff
@@ -219,7 +219,7 @@ interpreted literal. These are special cases of
 
 For more information about the allowable format of interpreted
 literals, refer to the "Syntax" section of the respective data types:
-[`DATE`](date.html#syntax), [`INTERVAL`](interval.html#syntax),
+[`DATE`](date.html#syntax), [`INTERVAL`](interval.html#syntax), [`TIME`](time.html#syntax),
 [`TIMESTAMP`/`TIMESTAMPTZ`](timestamp.html#syntax).
```
[the remark this answers is not in the document]
Comment (lhirata) [reply v2.0/sql-constants.md]: Done

File: _includes/sidebar-data-v2.0.json [hunk 6]
```diff
@@ -723,6 +723,12 @@
               "/${VERSION}/data-types.html"
             ]
           },
+          {
+            "title": "<code>INET</code>",
```
[the remark this answers is not in the document]
Comment (lhirata) [reply _includes/sidebar-data-v2.0.json]: Good catch - I'll update to be alphabetical.

**Comment by lhirata** [comment]
Thanks for the review @nvanbenschoten!

File: v2.0/inet.md [hunk 7]
```diff
@@ -1,6 +1,92 @@
...
+type `INET` or
+[coerced to](sql-expressions.html#explicit-type-coercions) type
+`INET`.
+
+`INET` constants can be expressed using the following formats:
+
+Format | Description
+-------|-------------
+IPv4 | Standard [RFC791](https://tools.ietf.org/html/rfc791)-specified format of 4 octets expressed individually in decimal numbers and separated by periods. Optionally, the address can be followed by a subnet mask.<br><br> Examples: `'190.0.0.0'`, `'190.0.0.0/24'`
+IPv6 | Standard [RFC8200](https://tools.ietf.org/html/rfc8200)-specified format of 8 colon-separated groups of 4 hexadecimal digits. An IPv6 address can be mapped to an IPv4 address. Optionally, the address can be followed by a subnet mask.<br><br> Example: `'2001:4f8:3:ba:2e0:81ff:fe22:d1f1'`, `'2001:4f8:3:ba:2e0:81ff:fe22:d1f1/120'`, `'::ffff:192.168.0.1/24'`
```
Comment (jseldess) [inline v2.0/inet.md]: `Example:` > `Examples:`

Comment (jseldess) [reply v2.0/inet.md]: nit: You also don't need a space after a `<br>`.

File: v2.0/inet.md [hunk 8]
```diff
@@ -1,6 +1,92 @@
...
+| registration_date | DATE   | true  | NULL    | {}          |
++-------------------+--------+-------+---------+-------------+
+~~~
+
+{% include copy-clipboard.html %}
+~~~ sql
+> INSERT INTO computers
+  VALUES
+    ('192.168.0.1', 'info@cockroachlabs.com', '2018-01-31'),
+    ('192.168.0.2/10', 'lauren@cockroachlabs.com', '2018-01-31');
```
Comment (jseldess) [inline v2.0/inet.md]: At end of line, change `;` to `,`; otherwise, this statement fails.

File: v2.0/inet.md [hunk 9]
```diff
@@ -1,6 +1,92 @@
...
+
+## Supported Casting & Conversion
+
+`INET` values can be [cast](data-types.html#data-type-conversions-casts) to the following data type:
+
+Type | Details
+-----|--------
+`STRING` | Converts to format `'Address/subnet'`.
+
+## See Also
```
Comment (jseldess) [inline v2.0/inet.md]: Please make these a bulleted list.

File: v2.0/inet.md (hunk 7 above)
Comment (lhirata) [reply v2.0/inet.md]: Done

File: v2.0/inet.md (hunk 8 above)
Comment (lhirata) [reply v2.0/inet.md]: Done

File: v2.0/inet.md (hunk 9 above)
Comment (lhirata) [reply v2.0/inet.md]: Done

File: v2.0/data-types.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (nvanbenschoten) [reply v2.0/data-types.md]: Are we going to remove the `New in v1.1` tags from the 2.0 docs?

**Comment by nvanbenschoten** [comment]
LGTM!

File: v2.0/data-types.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (lhirata) [reply v2.0/data-types.md]: I've been leaving the `1.1` flags in the 2.0. I think (if we remove them) we'll do it all at once.

**Pull request merged**